### PR TITLE
Fix power usage comparison script

### PR DESCRIPTION
**Pull request by petitfou**
Do not compute percentage as everage percentages of separate components Should compute the overall power usage and compute gain from this value

KIVU-147

### CODE REVIEW

File: checkbox-provider-kivu/bin/compare_power_consumption.py [hunk 1]
```diff
@@ -26,6 +26,17 @@ def load_energy(fname: str ) -> tuple[float, dict[str, float]]:
 
     return (duration, energy_usage)
 
+def get_energy_usage(rapl_output):
+    consumed_enegry = 0.0
+    p0 = rapl_output.get('package-0', None)
+    if p0:
+        consumed_enegry = p0
```
Comment (kissiel) [inline checkbox-provider-kivu/bin/compare_power_consumption.py:33]: aren't the 4 lines above equivalent to consumed_energy = rapl_output.get('package-0', 0.0)?

File: checkbox-provider-kivu/bin/compare_power_consumption.py [hunk 2]
```diff
@@ -26,6 +26,17 @@ def load_energy(fname: str ) -> tuple[float, dict[str, float]]:
...
+    # on some recent architectures, we have psys package that includes package-0
+    psys = rapl_output.get('psys', None)
```
Comment (kissiel) [inline checkbox-provider-kivu/bin/compare_power_consumption.py:35]: again s/None/0.0 makes more sense I guess

File: checkbox-provider-kivu/bin/compare_power_consumption.py [hunk 3]
```diff
@@ -34,27 +45,27 @@ def load_energy(fname: str ) -> tuple[float, dict[str, float]]:
     no_hwacc_energy_file = sys.argv[2]
     min_gain = float(sys.argv[3])
 
+    # data format
+    # Example: (600.097183, {'psys': 9976.218114, 'package-0': 3658.829745, 'core': 825.515099, 'uncore': 318.009806})
```
Comment (kissiel) [inline checkbox-provider-kivu/bin/compare_power_consumption.py:49]: if you have the realworld numbers, why not create a unit tests out of them?

File: checkbox-provider-kivu/bin/compare_power_consumption.py [hunk 4]
```diff
@@ -34,27 +45,27 @@ def load_energy(fname: str ) -> tuple[float, dict[str, float]]:
...
+    no_hwacc_cons = get_energy_usage(no_hwacc_energy[1])
+
+    print(f'Enabled: {hwacc_energy} - Consumption : {hwacc_cons}')
+    print(f'Disabled: {no_hwacc_energy} - Consumption : {no_hwacc_cons}')
 
-    avg_gain = statistics.mean(gain_energy.values())
+    hwacc_joule_per_sec = hwacc_cons / hwacc_energy[0]
+    no_hwacc_joule_per_sec = no_hwacc_cons / no_hwacc_energy[0]
+    if no_hwacc_joule_per_sec != 0:
+        gain = ((no_hwacc_joule_per_sec - hwacc_joule_per_sec) / no_hwacc_joule_per_sec) * 100
```
Comment (kissiel) [inline checkbox-provider-kivu/bin/compare_power_consumption.py:62]: This 100 and the -1 000 000 look totally arbitrary, I think they call for a comment.

File: checkbox-provider-kivu/bin/compare_power_consumption.py [hunk 5]
```diff
@@ -34,27 +45,27 @@ def load_energy(fname: str ) -> tuple[float, dict[str, float]]:
...
+    else:
+        gain = -1000000 * hwacc_joule_per_sec
```
Comment (kissiel) [inline checkbox-provider-kivu/bin/compare_power_consumption.py:64]: s/1000000/1_000_000

You're already using f-strings, so 3.6 python is required, meaning that you can use number literal separator.

**Review by kissiel — COMMENTED** [review]
Improvements ideas below.

File: checkbox-provider-kivu/bin/compare_power_consumption.py (hunk 3 above)
Comment (petitfou) [reply checkbox-provider-kivu/bin/compare_power_consumption.py]: > if you have the realworld numbers, why not create a unit tests out of them?

no, they are not real world numbers, just an example showing how data will look like

File: checkbox-provider-kivu/bin/compare_power_consumption.py (hunk 4 above)
Comment (petitfou) [reply checkbox-provider-kivu/bin/compare_power_consumption.py]: 100 is to made the gain to be a percentage, agree on 1000000, i will manage to remove it

**Review by mckees — CHANGES_REQUESTED** [review]
You have two separate commits here, but they both have the same title and commit message. Can you either squash them into one or give better commit messages?

Also nit on the commit message: *average

**Comment by petitfou** [comment]
> You have two separate commits here, but they both have the same title and commit message. Can you either squash them into one or give better commit messages?
> 
> Also nit on the commit message: *average

Ok

**Review by mckees — APPROVED** [review]

**Pull request merged**